InstantSearch.js/routing-seo-friendly/src/search-router.js

```diff
@@ -0,0 +1,91 @@
+/* global instantsearch */
+
+function getCategorySlug(name) {
+  return name.replace(/\s/g, '-');
+}
+
+function getCategoryName(slug) {
+  return slug.replace(/-/g, ' ');
+}
+
+const router = instantsearch.routers.history({
+  windowTitle({ category, q }) {
+    const query = q ? `Results for "${q}"` : 'Search';
+
+    if (category) {
+      return `${category} – ${query}`;
+    }
+
+    return query;
+  },
+  createURL({ qsModule, routeState, location }) {
+    const { protocol, hostname, port = '', hash } = location;
+    const portWithPrefix = port === '' ? '' : `:${port}`;
+    const pathname = routeState.category
+      ? `/${getCategorySlug(routeState.category)}/`
+      : '';
+    const queryParameters = {};
+
+    if (routeState.q) {
+      queryParameters.q = encodeURIComponent(routeState.q);
+    }
+    if (routeState.p !== 1) {
+      queryParameters.p = routeState.p;
+    }
+    if (routeState.brands) {
+      queryParameters.brands = encodeURIComponent(routeState.brands);
+    }
+
+    const queryString = qsModule.stringify(queryParameters, {
+      addQueryPrefix: true,
+    });
+
+    return `${protocol}//${hostname}${portWithPrefix}${pathname}${queryString}${hash}`;
+  },
+  parseURL({ qsModule, location }) {
+    const routeStateString = location.pathname.split('/')[1];
+    const category = getCategoryName(routeStateString.split('/')[0]);
+    const queryParameters = qsModule.parse(location.search.slice(1));
+    const { q = '', p, brands } = queryParameters;
+
+    return {
+      q: decodeURIComponent(q),
+      p,
+      category: decodeURIComponent(category),
+      brands,
+    };
+  },
+});
+
+const stateMapping = {
```

Comment:
This is still a lot of code and some of it isn't always easy to read.
I would suggest maybe we reduce the complexity of the example by 
- removing `q` for `query` and just keeping `query` everywhere
- removing `p` for `page` and just keeping `page` everywhere
- removing `brands` or any kind of refinement list
     * because I don't see people adding theses kind of links to their sitemaps  `mywebsite.com/phones/?brands=Apple|Samsung`, but I may be wrong
     * it removes this chunky part `uiState.refinementList && uiState.refinementList.brand && uiState.refinementList.brand.join('|')`

Reply:
Agreed on Slack to go for:

```
https://website.com/search/Cell+Phones/?query=m&page=4&brands=Metra&brands=Samsung
```

Let me know what you think.